Feature/pagination filtration sorting

AndrianovaVeronika opened the pull request.

Commit:
fix: interpolation

Commit:
Merge branch 'feature/internationalization' into develop

Commit:
feat: cities filtration form, AutocompleteField, RangeInputField, getFilteredCities (store)
fix: CitiesTable, FormikTextField,

Commit:
feat: Master-, user- orders filtration forms
fix: cities filtration form, AutocompleteField, RangeInputField

nik-clockwise commented on src/store/actions/users.js:
```diff
@@ -24,6 +26,7 @@ export const resetPassword = createAsyncThunk(users.RESET_PASSWORD, async (id, t
 
 export default {
     getAll: getUsers,
+    getFiltered: getFilteredUsers,
```
Обчно делают не обязательный параметр в метод all и не создают отдельный.
Еще в таком подходе непонятно откуда вообще берутся данные для фильтра, магия какая-то

AndrianovaVeronika replied:
Это просто объект, который принимает в себя actions. Так можно использовать в шаблонных компонентах.

nik-clockwise commented on src/store/initialState.js:
```diff
@@ -7,21 +7,27 @@ const initialState = {
     },
     orders: {
         ordersList: [],
+        ordersFilteredList: [],
```
Получается много лишнего/дублирующегося кода

AndrianovaVeronika replied:
Я не могу отнести это в один лист так как мне нужен отсортированный и неотсортированный одновременно

nik-clockwise replied:
А это как? Зачем нам все посты, если мы фильтруем посты? 2 запроса на бек, когда можно обойтись одним

nik-clockwise commented on src/store/selectors/citiesSelector.js:
```diff
@@ -1,4 +1,7 @@
 
 export const getCitiesSelector = createSelector((state) => state.cities, (state) => state.cities.citiesList);
+
+export const getFilteredCitiesSelector = createSelector((state) => state.cities, (state) => state.cities.citiesFilteredList);
```
Надо это всё зарефакторить что бы методы all принимали фильтр и всё работало ок

AndrianovaVeronika replied:
Это селекторы, они просто достают списки из стора. Для самих методов есть actionApi с методом GET_FILTERED, который возвращает функцию принимающую фильтры.

nik-clockwise replied:
Я так понимаю, что в апи должен быть один метод getCities() а принимать он удет фильтры и пагинационные параметры

Commit:
fix: global variables unnecessary use (citiesList, mastersList, etc)
feat: getters, getClockTypeById, MasterFiltrationForm, UserFiltrationForm, FormikDateRangeField, YesNoChooseField
delete: get filtered actions from store, errors
rename: FormikDataTableField -> FormikDateTimeField

Commit:
fix: styles, unused logs/variables, Filter forms, AutocompleteField
feat: onClean button of Filter forms

nik-clockwise commented on src/components/DataTables/CitiesTable/index.js:
```diff
@@ -25,21 +22,21 @@ const CitiesTable = () => {
-    const dispatch = useDispatch();
+    const [rows, setRows] = useState([]);
+    useEffect(async () => {
+        setRows(await getAllCities());
```
А тут все города выбираем, внутри есть пагинация? Выбирается часть городов из базы или все?

nik-clockwise replied:
Надо часть, т.к. все записи брать нельзя, когда их 1миллион то будет очень тяжело

nik-clockwise commented on src/components/DataTables/MastersTable/index.js:
```diff
@@ -35,21 +36,18 @@ const MastersTable = () => {
-    const dispatch = useDispatch();
+    const [rows, setRows] = useState([]);
+    useEffect(async () => {
+        setRows(await getAllMasters());
```
Вопрос тот же, что с количеством записей
